singlecellmultiomics/universalBamTagger/bamtagmultiome.py

```diff
@@ -0,0 +1,1229 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import pysam
+from itertools import chain
+from singlecellmultiomics.molecule import MoleculeIterator, ReadIterator
+import singlecellmultiomics
+import singlecellmultiomics.molecule
+from singlecellmultiomics.molecule.consensus import calculate_consensus
+import singlecellmultiomics.fragment
+from singlecellmultiomics.bamProcessing.bamFunctions import sorted_bam_file, get_reference_from_pysam_alignmentFile, write_program_tag, MapabilityReader, verify_and_fix_bam,add_blacklisted_region
+from singlecellmultiomics.utils import is_main_chromosome
+from singlecellmultiomics.utils.submission import submit_job
+import singlecellmultiomics.alleleTools
+from singlecellmultiomics.universalBamTagger.customreads import CustomAssingmentQueryNameFlagger, BulkFlagger
+import singlecellmultiomics.features
+from pysamiterators import MatePairIteratorIncludingNonProper, MatePairIterator
+from singlecellmultiomics.universalBamTagger.tagging import generate_tasks, prefetch, run_tagging_tasks, write_job_gen_to_bed
+from singlecellmultiomics.bamProcessing.bamBinCounts import blacklisted_binning_contigs,get_bins_from_bed_iter
+from singlecellmultiomics.utils.binning import bp_chunked
+from singlecellmultiomics.bamProcessing import merge_bams, get_contigs_with_reads, sam_to_bam
+from singlecellmultiomics.fastaProcessing import CachedFastaNoHandle
+from singlecellmultiomics.utils.prefetch import UnitialisedClass
+from multiprocessing import Pool
+from typing import Generator
+import argparse
+import uuid
+import os
+import sys
+import colorama
+import pkg_resources
+import pickle
+from datetime import datetime
+from time import sleep
+# Supress [E::idx_find_and_load] Could not retrieve index file for, see https://github.com/pysam-developers/pysam/issues/939
+pysam.set_verbosity(0)
+
+argparser = argparse.ArgumentParser(
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    description='Assign molecules, set sample tags, set alleles')
+argparser.add_argument('bamin', type=str, help='Input BAM file, SAM files can also be supplied but will be converted to BAM')
+argparser.add_argument(
+    '-ref',
+    type=str,
+    default=None,
+    help="Path to reference fast (autodected if not supplied)")
+
+argparser.add_argument('--bamfile', type=str, help="output bam file", required=True)
```

Comment:
The only change I made to integrate the tool in our subcommand: `--bamfile`, `-o`, and `-method`. The other code of `singlecellmultiomics` is kept as-is to minimized maintenance effort from our side.

Reply:
That being said. Feel free to ignore reviewing of `singlecellmultiomics` but this code snippet.